src/oracle/monitor.js

```diff
@@ -0,0 +1,446 @@
+    return latestPrice;
+  } catch (err) {
+    console.log("Price could not be obtained");
+    return 0;
+  }
+};
+
+/**
+ * Function to get offers and balances
+ * @param {*} follower offers and balances
+ * @param {*} oracle oracle address
+ * @returns an object containing the offers and balances
+ */
+const getOffersAndBalances = async (follower, oracle) => {
+  let toReturn = {
+    offers: [],
+    balances: [],
+  };
+  let counter = 0;
+  let lastVisited = 0;
+
+  for await (const followerElement of iterateReverse(follower)) {
+    if (counter === 10) {
+      break;
+    }
+
+    // If it is an offer status
+    if (followerElement.value.updated === "offerStatus") {
+      // Get id
+      let id = followerElement.value.status.id;
+
+      // If a new and final state
+      if (id !== lastVisited) {
+        // If it is not failed
+        if (!followerElement.value.status.hasOwnProperty("error")) {
+          toReturn["offers"].push(followerElement.value);
+          counter++;
+        }
+        lastVisited = id;
+      }
+    }
+  }
+
+  // Get current purses
+  let current = await getCurrent(oracle, fromBoard, { vstorage });
+  for (let i = 0; i < current.purses.length; i++) {
+    let currentPurse = current.purses[i];
+    toReturn["balances"].push(currentPurse.balance);
+  }
+
+  return toReturn;
+};
+
+/**
+ * Function to get latest prices for oracle
+ * @param {*} oracle oracle address
+ * @param {*} oracleDetails oracle details
+ * @param {*} state oracle's latest state
+ * @returns last results including the oracle submitted price
+ */
+export const getLatestPrices = async (oracle, oracleDetails, state) => {
+  // Get feeds for oracle
+  let feeds = oracleDetails["feeds"];
+  console.log("Getting prices for", oracle, feeds);
+
+  const unserializer = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+  const leader = makeLeader(networkConfig.rpcAddrs[0]);
+
+  const follower = await makeFollower(`:published.wallet.${oracle}`, leader, {
+    // @ts-expect-error xxx
+    unserializer,
+  });
+
+  let offersBalances = await getOffersAndBalances(follower, oracle);
+
+  // Get last offer id from offers from state
+  let lastOfferId = isNaN(state["last_offer_id"]) ? 0 : state["last_offer_id"];
+
+  // Initialise variable to hold results
+  let lastResults = {
+    last_offer_id: lastOfferId,
+    values: state["values"] ? state["values"] : {},
+  };
+
+  // Loop through offers starting from last visited index
+  for (let i = 0; i < offersBalances.offers.length; i++) {
+    // Get current offer
+    let currentOffer = offersBalances.offers[i];
+    let id = Number(currentOffer["status"]["id"]);
+
+    // If we found the last visited offer id in previous check, stop looping
+    console.log("lastOfferId", lastOfferId, "currentId", id);
+    if (id <= lastOfferId) {
+      break;
+    }
+
+    // If a price invitation
+    if (
+      currentOffer["status"]["invitationSpec"]["invitationMakerName"] ==
```

Comment:
```suggestion
      currentOffer["status"]["invitationSpec"]["invitationMakerName"] ===
```

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/54132c2dc20f860fa9b36d27d57841b5ae869cdb